### DISCUSSION TITLE
How do I enable the Android Connection Priority when connecting to a BLE device?

### DISCUSSION THREAD

**Discussion by powerteqtpitman** [shinyorg/shiny#1446] [Q&A] [answered]
I know the 3.x documentation is still marked as beta, so if this isn't implemented yet, please let me know and I will wait.

I am looking at the shiny documentation for the peripheral and it says to use this code to change the connection priority for Android:

// connect
peripheral.Connect(new GattConnectionConfig
{
    /// <summary>
    /// This will cause disconnected peripherals to try to immediately reconnect.  It will cause WillRestoreState to fire on iOS. Defaults to true
    /// </summary>
    public bool IsPersistent { get;  set; } = true;

    /// <summary>
    /// Android only - If you have characteristics where you need faster replies, you can set this to high
    /// </summary>
    public ConnectionPriority Priority { get; set; } = ConnectionPriority.Normal;
});

I can't seem to find the GattConnectionConfig and the cross platform ConnectionConfig doesn't have the Priority property.

Please advise...

**Comment by aritchie** [comment, marked as answer]
```csharp
IPeripheral peripheral; // discovered peripheral
#if ANDROID
peripheral.Connect(new AndroidConnectionConfig(
    ConnectionPriority = Android.Bluetooth.GattConnectionPriority.High
));
#endif
```

**Reply by powerteqtpitman** [reply]
Thank you! I didn't realize I would have to put platform specific code.

I appreciate the quick response.
